### PR TITLE
Dyn 2586 - Change the tests on ViewExtensionLibraryTests.LibraryResourceProviderTests

### PR DESCRIPTION
**Pull request by alfredo-pozo**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

MInimize the apperance of this failed tests 

ViewExtensionLibraryTests.LibraryResourceProviderTests.ParallelEventObserver 
ViewExtensionLibraryTests.LibraryResourceProviderTests.LibraryDataUpdatedEventRaised

So we are disposing of some instances (AutoEventReset type) that are recommended by MSFT

https://jira.autodesk.com/browse/DYN-2586

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 
@mjkkirschner

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@alfredo-pozo looks good, can you confirm it makes a difference or it's just good practice so why not do it?

**Comment by alfredo-pozo** [comment]
I've just a made a test on SelfServe https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/master/27 with the changes,  and the previous one, failed on one of the test that I've changed, so I think the changes can make a difference.

The changes were made according to this information on MSFT site https://docs.microsoft.com/en-us/dotnet/api/system.threading.autoresetevent?view=netframework-4.7 , please check the "Important" sections of that page

**Comment by QilongTang** [comment]
@alfredo-pozo seems the test still failed on top of your change in this PR?

**Comment by alfredo-pozo** [comment]
Yes, but is a different test on another TestSuite that I didn't change, I'll check what could be the reason for this one https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/207/testReport/junit/ViewExtensionLibraryTests/LibraryViewControllerTests/InstantiatingLazyLoadedCustomNodeShouldNotRaiseUpdate/

File: test/ViewExtensionLibraryTests/LibraryResourceProviderTests.cs [hunk 1]
```diff
@@ -553,14 +557,15 @@ public void RefireThrottledEvents()
             var result = Parallel.ForEach(list, x => observer.OnEvent(x));
 
             resetevent.WaitOne(250);
-            Assert.IsTrue(result.IsCompleted);
+			Assert.IsTrue(result.IsCompleted);
```
Comment (QilongTang) [inline test/ViewExtensionLibraryTests/LibraryResourceProviderTests.cs:560]: Can you fix this line as well?

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**